features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/notificationsettings/RoomNotificationSettingsNode.kt

```diff
@@ -48,10 +56,18 @@ class RoomNotificationSettingsNode @AssistedInject constructor(
     @Composable
     override fun View(modifier: Modifier) {
         val state = presenter.present()
-        RoomNotificationSettingsView(
-            state = state,
-            modifier = modifier,
-            onBackPressed = this::navigateUp,
-        )
+        if(inputs.showUserDefinedSettingStyle) {
```

Comment:
I think the pattern we adopted with appyx tends to associate one presenter with one view only.
This is because we can't easily test the Node class so any logic (like this `if``) you put in the Node class will likely not be covered by unit tests.

A better IMHO solution would have been to pass `inputs.showUserDefinedSettingStyle` as input param to `RoomNotificationSettingsPresenter` and then expose it through the state class.
Then create a super-composable function that holds this if logic.
This way we could cover it with screenshot tests at least.

@ganfra what do you think?

Anyhow, not a blocker right now, I just wanted to share this with others to better understand the usage of our patterns.

Reply:
Makes sense, moved it to down to view.